### ISSUE TITLE
experiment details

### ISSUE THREAD

**Issue by SagacityPuppy** [qq456cvb/KdNet-Tensorflow#1]
Hi, thanks for your sharing, could you tell me the about the hardware configuration and the time cost of 1-epoch training? In addition, is the base tensorflow enviroment is 'tf-gpu'? I will be very appreciate for your reply!

**Comment by qq456cvb** [comment]
Yes, it uses GPU TensorFlow (TF 1.x with Tensorpack), so a `tf-gpu` environment is what you want.

I don't have the exact original hardware / per-epoch time noted, but a few useful points: the network itself is small (feature widths up to 512, batch size 32, 300 epochs on ModelNet40 with ~9840 training samples), so the GPU compute per epoch is light. The main cost is that the kd-tree is rebuilt on the fly on the CPU for every sample (`KDTree.py`), which is usually the bottleneck, so epoch time depends more on CPU / dataflow throughput than on the GPU. On a typical single modern GPU it trains comfortably; if epochs feel slow, increase the Tensorpack dataflow workers to parallelize the kd-tree construction.
